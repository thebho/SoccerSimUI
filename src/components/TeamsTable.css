.Table {
  display: flex;
}

.Table-Col-Nam {
  width: 160px;
}

.Table-Col {
  width: 60px;
  padding: 0px;
  padding-left: 0px;
  padding-right: 0px;
}
