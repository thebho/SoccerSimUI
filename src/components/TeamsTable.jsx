//@flow
import React from 'react';
import type { Team } from '../model';
import {
  Table,
  TableBody,
  TableHeader,
  TableHeaderColumn,
  TableRow,
  TableRowColumn,
} from 'material-ui/Table';

type Props = {
  teams: Array<Team>,
};

const TeamsTable = (props: Props): typeof Table => (
  <Table
    displaySelectAll={false}
    adjustForCheckbox={false}
  >
    <TableHeader
      adjustForCheckbox={false}
      displaySelectAll={false}
    >
      <TableRow>
        <TableHeaderColumn>Team</TableHeaderColumn>
        <TableHeaderColumn>Wins</TableHeaderColumn>
        <TableHeaderColumn>Draws</TableHeaderColumn>
        <TableHeaderColumn>Losses</TableHeaderColumn>
      </TableRow>
    </TableHeader>
    <TableBody
      displayRowCheckbox={false}
    >
    {props.teams.map((team) => (
      <TableRow
        key={team.name}
      >
        <TableRowColumn>{team.name}</TableRowColumn>      <TableRowColumn>{team.gamesWon}</TableRowColumn>
        <TableRowColumn>{team.gamesDrawn}</TableRowColumn>      <TableRowColumn>{team.gamesLost}</TableRowColumn>
      </TableRow>
    ))}
    </TableBody>
  </Table>
);

export default TeamsTable;
