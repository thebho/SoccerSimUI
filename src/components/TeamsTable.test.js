import React from 'react';
import ReactDOM from 'react-dom';
import { TeamsTable } from './TeamsTable';
import { shallow, mount, render } from 'enzyme';
import toJson from 'enzyme-to-json';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';

const mockTeams = [
  {
    name: 'Arsenal',
    gamesWon: 2,
    gamesLost: 0,
    gamesDrawn: 0,
    goalsScored: 20,
    goalsAllowed: 0,
    Abv: 'ARS',
    TeamID: 'ARS',
  },
  {
    name: 'Sp*rs',
    gamesWon: 0,
    gamesLost: 2,
    gamesDrawn: 0,
    goalsScored: 0,
    goalsAllowed: 20,
    Abv: 'TOT',
    TeamID: 'TOT',
  },
];
describe('TeamsTable', () => {
  // it('matches snapshot with mock teams', () => {
  //   const wrapper = shallow(
  //     <MuiThemeProvider>
  //       <TeamsTable teams={mockTeams} />
  //     </MuiThemeProvider>
  //   ).dive();
  //   expect(toJson(wrapper)).toMatchSnapshot();
  // });
  // it('renders without teams', () => {
  //   const wrapper = shallow(
  //     <MuiThemeProvider>
  //       <TeamsTable />
  //     </MuiThemeProvider>
  //   ).dive();
  //   expect(toJson(wrapper)).toMatchSnapshot();
  // });
});
