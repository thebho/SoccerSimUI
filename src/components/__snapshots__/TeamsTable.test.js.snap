// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TeamsTable matches snapshot with mock teams 1`] = `
<Table
  adjustForCheckbox={false}
  allRowsSelected={false}
  className="Table"
  displaySelectAll={false}
  fixedFooter={true}
  fixedHeader={true}
  height="inherit"
  multiSelectable={false}
  selectable={true}
>
  <TableHeader
    adjustForCheckbox={false}
    displaySelectAll={false}
    enableSelectAll={true}
    selectAllSelected={false}
  >
    <TableRow
      displayBorder={true}
      hoverable={false}
      hovered={false}
      selectable={true}
      striped={false}
      style={
        Object {
          "height": "25px",
          "paddingBottom": "0px",
          "paddingLeft": "10px",
          "paddingRight": "10px",
          "paddingTop": "0px",
          "textAlign": "center",
          "width": "40px",
        }
      }
    >
      <TableHeaderColumn
        className="Table-Col-Nam"
      >
        Team
      </TableHeaderColumn>
      <TeamHeaderColumnDefault
        text="GP"
      />
      <TeamHeaderColumnDefault
        text="W"
      />
      <TeamHeaderColumnDefault
        text="D"
      />
      <TeamHeaderColumnDefault
        text="L"
      />
      <TeamHeaderColumnDefault
        text="GF"
      />
      <TeamHeaderColumnDefault
        text="GA"
      />
                
      <TeamHeaderColumnDefault
        text="GD"
      />
      <TeamHeaderColumnDefault
        text="Points"
      />
    </TableRow>
  </TableHeader>
  <TableBody
    allRowsSelected={false}
    deselectOnClickaway={true}
    displayRowCheckbox={false}
    multiSelectable={false}
    preScanRows={true}
    selectable={true}
    style={Object {}}
  >
    <TableRow
      displayBorder={true}
      hoverable={false}
      hovered={false}
      key="ARS"
      selectable={true}
      striped={false}
      style={
        Object {
          "height": "25px",
        }
      }
    >
      <TableRowColumn
        className="Table-Col-Nam"
        hoverable={false}
        style={
          Object {
            "height": "25px",
          }
        }
      />
      <TeamColumnDefault
        text={NaN}
      />
      <TeamColumnDefault />
      <TeamColumnDefault />
      <TeamColumnDefault />
      <TeamColumnDefault />
      <TeamColumnDefault />
      <TeamColumnDefault
        text={NaN}
      />
      <TeamColumnDefault
        text={NaN}
      />
    </TableRow>
    <TableRow
      displayBorder={true}
      hoverable={false}
      hovered={false}
      key="TOT"
      selectable={true}
      striped={false}
      style={
        Object {
          "height": "25px",
        }
      }
    >
      <TableRowColumn
        className="Table-Col-Nam"
        hoverable={false}
        style={
          Object {
            "height": "25px",
          }
        }
      />
      <TeamColumnDefault
        text={NaN}
      />
      <TeamColumnDefault />
      <TeamColumnDefault />
      <TeamColumnDefault />
      <TeamColumnDefault />
      <TeamColumnDefault />
      <TeamColumnDefault
        text={NaN}
      />
      <TeamColumnDefault
        text={NaN}
      />
    </TableRow>
  </TableBody>
</Table>
`;

exports[`TeamsTable renders without teams 1`] = `
<Table
  adjustForCheckbox={false}
  allRowsSelected={false}
  className="Table"
  displaySelectAll={false}
  fixedFooter={true}
  fixedHeader={true}
  height="inherit"
  multiSelectable={false}
  selectable={true}
>
  <TableHeader
    adjustForCheckbox={false}
    displaySelectAll={false}
    enableSelectAll={true}
    selectAllSelected={false}
  >
    <TableRow
      displayBorder={true}
      hoverable={false}
      hovered={false}
      selectable={true}
      striped={false}
      style={
        Object {
          "height": "25px",
          "paddingBottom": "0px",
          "paddingLeft": "10px",
          "paddingRight": "10px",
          "paddingTop": "0px",
          "textAlign": "center",
          "width": "40px",
        }
      }
    >
      <TableHeaderColumn
        className="Table-Col-Nam"
      >
        Team
      </TableHeaderColumn>
      <TeamHeaderColumnDefault
        text="GP"
      />
      <TeamHeaderColumnDefault
        text="W"
      />
      <TeamHeaderColumnDefault
        text="D"
      />
      <TeamHeaderColumnDefault
        text="L"
      />
      <TeamHeaderColumnDefault
        text="GF"
      />
      <TeamHeaderColumnDefault
        text="GA"
      />
                
      <TeamHeaderColumnDefault
        text="GD"
      />
      <TeamHeaderColumnDefault
        text="Points"
      />
    </TableRow>
  </TableHeader>
  <TableBody
    allRowsSelected={false}
    deselectOnClickaway={true}
    displayRowCheckbox={false}
    multiSelectable={false}
    preScanRows={true}
    selectable={true}
    style={Object {}}
  />
</Table>
`;
